cea/optimization/slave/cooling_resource_activation.py

```diff
@@ -148,145 +148,266 @@ def cooling_resource_activator(Q_thermal_req,
+                and Q_cooling_unmet_W >= Qc_output_BaseVCC_WS_max_W:
+            BaseVCC_WS_activated = True
+            Qc_BaseVCC_WS_gen_directload_W = Qc_output_BaseVCC_WS_max_W
             Qc_BaseVCC_WS_gen_storage_W = 0.0
-            Qc_from_storage_W, Q_DailyStorage_content_W = \
-                daily_storage_class.discharge_storage(Q_cooling_unmet_W - min(size_WS_BaseVCC_W, Q_therm_water_body_W))
-            Q_BaseVCC_WS_gen_W = Q_BaseVCC_WS_gen_directload_W + Qc_BaseVCC_WS_gen_storage_W
-        else:
-            Q_BaseVCC_WS_gen_directload_W = Q_cooling_unmet_W
-            Qc_BaseVCC_WS_gen_storage_W, Q_DailyStorage_content_W = \
-                daily_storage_class.charge_storage(min(size_WS_BaseVCC_W, Q_therm_water_body_W) - Q_cooling_unmet_W)
+            Qc_from_storage_W, Qc_DailyStorage_content_W = \
+                daily_storage_class.discharge_storage(Q_cooling_unmet_W - Qc_output_BaseVCC_WS_max_W)
+            Qc_BaseVCC_WS_gen_W = Qc_BaseVCC_WS_gen_directload_W + Qc_BaseVCC_WS_gen_storage_W
+
+        # Activation Case 2: The water temperature doesn't allow for free cooling, therefore the VCC is activated.
+        # The maximum VCC output is larger than the unmet cooling demand, therefore the storage is charged.
+        elif T_district_cooling_supply_K - T_source_average_water_body_K < DT_COOL \
+                and Q_cooling_unmet_W < Qc_output_BaseVCC_WS_max_W:
+            BaseVCC_WS_activated = True
+            Qc_BaseVCC_WS_gen_directload_W = Q_cooling_unmet_W
+            Qc_BaseVCC_WS_gen_storage_W, Qc_DailyStorage_content_W = \
+                daily_storage_class.charge_storage(Qc_output_BaseVCC_WS_max_W - Q_cooling_unmet_W)
             Qc_from_storage_W = 0.0
-            Q_BaseVCC_WS_gen_W = Q_BaseVCC_WS_gen_directload_W + Qc_BaseVCC_WS_gen_storage_W
-        # If water body temperature is higher than the temperature required in the district cooling network's heat
-        # exchanger, use the vapour compression chiller to bridge the temperature difference...
-        if (T_district_cooling_supply_K - DT_COOL) < T_source_average_water_body_K:
-            WS_BaseVCC_capacity = master_to_slave_variables.WS_BaseVCC_size_W
-            Q_BaseVCC_WS_gen_W, \
-            E_BaseVCC_WS_req_W = calc_vcc_operation(Q_BaseVCC_WS_gen_W,
+            Qc_BaseVCC_WS_gen_W = Qc_BaseVCC_WS_gen_directload_W + Qc_BaseVCC_WS_gen_storage_W
+
+        # Activation Case 3: The water temperature allows for free cooling, therefore the VCC is bypassed.
+        # The unmet cooling demand is larger than the water body's cooling potential, hence the storage is discharged.
+        elif T_district_cooling_supply_K - T_source_average_water_body_K >= DT_COOL \
+                and Q_cooling_unmet_W >= Qc_water_body_remaining_W:
+            FreeCooling_WS_activated = True
+            Qc_FreeCooling_WS_directload_W = Qc_water_body_remaining_W
+            Qc_water_body_to_storage_W = 0.0
```

Comment:
This value is declared as 0.0 and I missed to find where it changes (since it wouldn't make sense to remain 0 throughout the loops). Can you point me to where this happens?

Reply:
These few lines of code describe the activation case, where water from the water body is used directly to cover the cooling demand (free cooling) but the cooling capacity of the water body is large enough to cover the demand (i.e. the there is no leftover capacity for storage). If this activation case is applied for a specific time step, the value does not change in the rest of the activation chain (the cooling capacity of the water body is used up for the time step). 
However, in the next time step the whole function get's called again and a different activation case might apply.